dev-lang/zig/zig-0.9.1.ebuild

```diff
@@ -21,26 +21,24 @@ SLOT="0"
 IUSE="test +threads"
 RESTRICT="!test? ( test )"
 
-PATCHES=("${FILESDIR}/${P}-fix-single-threaded.patch")
+PATCHES=(
+	"${FILESDIR}/${P}-fix-single-threaded.patch"
+	"${FILESDIR}/${P}-fix-riscv.patch"
```

Comment:
I tried to build this on riscv but I still had some issues, although given it's merged upstream we might leave it

Reply:
Is this the same problem as without the patch, or is it different?

Reply:
I think it was different, but can't remember off the top of my head what it was, so I'll have to try building again with and without the patch.

Reply:
I'm trying to reproduce this with riscv chroot and QEMU user emulation, is it complete enough? (QEMU system emulation is slow for me & I don't have RISC-V hardware)

Reply:
Well, with this patch `zig` can be builded succesfully, but we can't build ncdu on RISC-V systems yet :(
```sh
$ /tmp/portage/dev-lang/zig-0.9.1/work/zig-0.9.1/build/zig build -Drelease-fast

ld.lld: error: /usr/lib64/lp64d/crt1.o:(.text+0x0): relocation R_RISCV_ALIGN requires unimplemented linker relaxation; recompile with -mno-relax
error: LLDReportedFailure
ncdu...The following command exited with error code 1:
/tmp/portage/dev-lang/zig-0.9.1/work/zig-0.9.1/build/zig build-exe /tmp/portage/sys-fs/ncdu-2.1.2/work/ncdu-2.1.2/src/main.zig /tmp/portage/sys-fs/ncdu-2.1.2/work/ncdu-2.1.2/src/ncurses_refs.c -lc -lncursesw -ltinfow -OReleaseFast --cache-dir /tmp/portage/sys-fs/ncdu-2.1.2/work/ncdu-2.1.2/zig-cache --global-cache-dir /root/.cache/zig --name ncdu -I /usr/include/ncursesw -D _DEFAULT_SOURCE -D _XOPEN_SOURCE=600 --enable-cache
error: the following build command failed with exit code 1:
/tmp/portage/sys-fs/ncdu-2.1.2/work/ncdu-2.1.2/zig-cache/o/23b40c8e0866163e938bfce718fc22b7/build /tmp/portage/dev-lang/zig-0.9.1/work/zig-0.9.1/build/zig /tmp/portage/sys-fs/ncdu-2.1.2/work/ncdu-2.1.2 /tmp/portage/sys-fs/ncdu-2.1.2/work/ncdu-2.1.2/zig-cache /root/.cache/zig -Drelease-fast
```

IMO that's because `zig` misdetects ABI:
```sh
$ /tmp/portage/dev-lang/zig-0.9.1/work/zig-0.9.1/build/zig targets | grep triple

"triple": "riscv64-linux.5.18.15...5.18.15-musl",
```

According to https://maskray.me/blog/2022-07-10-riscv-linker-relaxation-in-lld:
> On 2022-07-07, I added a RISC-V linker relaxation framework in ld.lld and implemented R_RISCV_ALIGN/R_RISCV_CALL/R_RISCV_CALL_PLT relaxation. The changes will be included in the next llvm-project release 15.0.0. This post describes the implementation.

Zig 0.10.0 will be self-hosted by default (see https://github.com/ziglang/zig/pull/12368) and possibly upgraded to LLVM 15, so [I think] we should postpone this until next release...